src/pineko/theory.py

```diff
@@ -409,9 +417,12 @@ def fk(self, name, grid_path, tcard, pdf):
         # loading ekos to produce a tmp copy
         with eko.EKO.read(eko_filename) as operators:
             # Skip the computation of the fktable if the eko is empty
-            if len(operators.mu2grid) == 0:
-                raise EmptyEkoError
-            eko_tmp_path = operators.paths.root.parent / "eko-tmp.tar"
+            if len(operators.mu2grid) == 0 and check.is_num_fonll(tcard["FNS"]):
+                rich.print(f"[green] Skipping empty eko for nFONLL.")
+                return
+            eko_tmp_path = (
+                operators.paths.root.parent / f"eko-tmp-{np.random.rand()}.tar"
```

Comment:
and if you want to be more diverse (for parallel, I guess), maybe put the grid name or dataset name somewhere

Reply:
No it is not for parallel. This is to avoid rotating (in x space) the original eko because, as @giacomomagni noticed, once rotated the eko is much heavier. So we copy the eko here and then we remove it once the fktable is done. For this reason I don't believe the eko name matters so much.

Reply:
Just to be precise, this is not for parallel because the FKtable computation is so fast that I don't believe having the same name at the same moment is something possible. Anyway putting the grid name could be a good idea

Reply:
This was indeed for the rotation memory problem, but the name fix is for parallel. 
In fact if you are computing many `FKtables` at the same time, it's safer/correct to have different temp file names. 
For the actual name any name it's fine for me.

Reply:
So, do you like how it is now?